src/ui/internal/hooks/useModalStates.ts

```diff
@@ -25,12 +25,16 @@ export const useModalStates = ({
   const [error, setError] = useState<string | null>(null);
   const [success, setSuccess] = useState<string | null>(null);
 
-  const onModalClose = () => {
+  const onModalClose = (reason: "close" | "change" = "close") => {
     if (!(error || success)) {
       const noActionCloseError = {
-        message: "User closed modal without taking any actions",
+        message:
+          reason === "change"
+            ? ""
+            : "User closed modal without taking any actions",
```

Comment:
`!reason && "User closed modal without taking any actions"` shorter, but would set it to `undefined` iso of `""`. Feasible here?

Reply:
Otherwise fine with your changes :)

Reply:
it's better log a "" empty in the dev tools than undefined just not to make the user confused.